### ISSUE TITLE
Inconsistencies between run()/output()/read()/read_err()

### ISSUE THREAD

**Issue by rickvanprim** [matklad/xshell#57]
Currently `run()` ignores `stdin()` and `output()`/`read()`/`read_err()` ignore `quiet()` (as it's always implied).  It seems like either the documentation should get updated to reflect this, or the functionality made consistent.

Proposing:

`run()` method
```rust
    pub fn run(&self) -> Result<()> {
        if !self.data.quiet {
            eprintln!("$ {}", self);
        }
        let mut command = self.to_command();
        let status = command.status().map_err(|err| Error::new_cmd_io(self, err))?;
        self.check_status(status)?;
        Ok(())
    }
```
becomes
```rust
    pub fn run(&self) -> Result<()> {
        self.output_impl(false, false).map(|_| ())
    }
```

`output_impl()` method adds
```rust
        if !self.data.quiet {
            eprintln!("$ {}", self);
        }
```

**Comment by domenicquirl** [comment]
Your suggestion seems reasonable to me. I'd like to add that in some cases things that currently are the same across these methods can be surprising as well because the methods have different use cases. 

As an example, I originally did not expect `output()` to fail (return `Err`) on non-zero exit codes. After all, it is the only option available if I want to look at precisely that exit code through the returned `Output`. However, without `.ignore_status(true)` it is currently only ever possible to see status code `0` in `Ok` results - which feels unhelpful, even if it matches the other methods on `Command`.

**Comment by matklad** [comment]
See also https://github.com/matklad/xshell/issues/63#issuecomment-1630669289

**Comment by blinsay** [comment]
Just lost some time to `read` not echoing a command the way `run` does. I can work around it, but +1 for the option to have xshell echo the command.

(ps - this crate is fantastic, thanks for writing it!)

**Comment by matklad** [comment]
Fixed, or at least changed sufficiently in 0.3.0! Not sure if I am fully happy with the new API, but I do like it better!

**Comment by woojiq** [comment]
Is it possible to achieve the following in 0.3.0pre2:

```rust
let remotes = cmd!(sh, "git remote").read().unwrap();
```

I want to have command output in `remotes` variable, but also see the command and output in the terminal (as if I had executed this command manually):
```bash
$ git remote
origin
upstream
```

I'd expect to see `read_echo` command that prints command and stdout. The library is amazing though.

_sorry for necroposting_
